libfeed/feed-store-provider.c

```diff
@@ -0,0 +1,1239 @@
+                                                                     g_hash_table_lookup (table, "ekn_id"),
+                                                                     duration,
+                                                                     thumbnail_stream,
+                                                                     eos_discovery_feed_knowledge_app_proxy_get_desktop_id (data->ka_proxy),
+                                                                     eos_discovery_feed_knowledge_app_proxy_get_bus_name (data->ka_proxy),
+                                                                     eos_discovery_feed_knowledge_app_proxy_get_knowledge_search_object_path (data->ka_proxy),
+                                                                     eos_discovery_feed_knowledge_app_proxy_get_knowledge_app_id (data->ka_proxy));
+      g_ptr_array_add (orderable_stores,
+                       eos_discovery_feed_orderable_model_new (EOS_DISCOVERY_FEED_BASE_CARD_STORE (store),
+                                                               data->type,
+                                                               eos_discovery_feed_knowledge_app_proxy_get_desktop_id (data->ka_proxy)));
+    }
+
+    return g_steal_pointer (&orderable_stores);
+}
+
+static gpointer
+append_discovery_feed_video_from_proxy (EosDiscoveryFeedKnowledgeAppProxy  *ka_proxy,
+                                        gpointer                            proxy_data,
+                                        GCancellable                       *cancellable,
+                                        GError                            **error)
+{
+  g_autoptr(VideoCardsFromShardsAndItemsData) marshal_data =
+    video_cards_from_shards_and_items_data_new (ka_proxy,
+                                                EOS_DISCOVERY_FEED_CARD_STORE_TYPE_VIDEO_CARD);
+  GDBusProxy *dbus_proxy = eos_discovery_feed_knowledge_app_proxy_get_dbus_proxy (ka_proxy);
+  const gchar *desktop_id = eos_discovery_feed_knowledge_app_proxy_get_desktop_id (ka_proxy);
+
+  return call_dbus_proxy_and_construct_from_models_and_shards (dbus_proxy,
+                                                               "GetVideos",
+                                                               desktop_id,
+                                                               video_cards_from_shards_and_items,
+                                                               marshal_data,
+                                                               cancellable,
+                                                               error);
+}
+
+typedef struct _ArtworkCardsFromShardsAndItemsData
+{
+  EosDiscoveryFeedKnowledgeAppProxy                *ka_proxy;
+  EosDiscoveryFeedCardLayoutDirection               direction;
+  EosDiscoveryFeedCardStoreType                     type;
+  guint                                             thumbnail_size;
```

Comment:
Shouldn't be any need for all these properties, its the same for all artwork cards. We can just pass the KA proxy directly.

Reply:
(Note that the comment doesn't apply here - for "article" cards these properties can vary depending on where the card is in the layout and whether it is a news card or an article card).